src/ArmTemplates/Common/API/Clients/Product/ProductsClient.cs

```diff
@@ -10,19 +10,32 @@
 using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Product.Responses;
 using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Constants;
 using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.ProductApis;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Products;
 using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
 
 namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Product
 {
     public class ProductsClient : ApiClientBase, IProductsClient
     {
-        const string GetAllApiProductsRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/apis/{4}/products?api-version={5}";
+        const string GetAllProductsLinkedToApiRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/apis/{4}/products?api-version={5}";
+        const string GetAllProductsRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/products?api-version={4}";
+
+        public async Task<List<ProductsTemplateResource>> GetAllAsync(ExtractorParameters extractorParameters)
+        {
+            (string azToken, string azSubId) = await this.Auth.GetAccessToken();
```

Comment:
Could we move this to the constructor of the base class instead of calling it in every client?

Reply:
This is not a one-time thing to be done - it is a call for getting Azure bearer authentication token and it has to be done right before the call.

If we place it in constructor, we can fall into a trouble. Imagine:
- you have 1000 extractors, each are making a request to Azure and it is lasting for a long time (maybe some Azure region is in trouble)
- you have initialized a bearer token for every client and saved it in each client
- once you reach i.e. 600th extractor your token has expired. How could you renew it?

Reply:
I guess it depends on how long the bearer token lasts and how long the program runs? If it expires before the application finishes then yeah it's a problem.

Alternatively, It could be static in the base class and we could just regenerate the token in the base class when it expires. 😄 

We don't have to do it, it's just an idea. We call this method everywhere with no params, so it'd be nice to hide it somewhere, but it's probably not be worth the effort.